src/main/java/org/broadinstitute/hellbender/utils/codecs/GENCODE/GencodeGtfCodec.java

```diff
@@ -0,0 +1,376 @@
+                        break;
+                }
+
+                ++currentLineNum;
+            }
+
+            // Increment our iterator here so we don't accidentally miss any features from the following gene
+            lineIterator.next();
+        }
+
+        // Do we have some records leftover that we need to aggregate into a feature:
+        if ( (gene != null) &&
+                (exonStore.size() != 0) || (leafFeatureStore.size() != 0) ) {
+
+            aggregateRecordsIntoGeneFeature(gene, transcript, exonStore, leafFeatureStore);
+            decodedFeature = gene;
+        }
+
+        // If we have other records left over we should probably yell a lot,
+        // as this is bad.
+        //
+        // However, this should never actually happen.
+        //
+        if ( (exonStore.size() != 0) || (leafFeatureStore.size() != 0) ) {
+
+            if (exonStore.size() != 0) {
+                logger.error("Gene Feature Aggregation: Exon store not empty: " + exonStore.toString());
+            }
+
+            if (leafFeatureStore.size() != 0) {
+                logger.error("Gene Feature Aggregation: leaf feature store not empty: " + leafFeatureStore.toString());
+            }
+
+            String msg = "Aggregated data left over after parsing complete: Exons: " + exonStore.size() + " ; LeafFeatures: " + leafFeatureStore.size();
+            throw new RuntimeException(msg);
+        }
+
+        return decodedFeature;
+    }
+
+    /**
+     * Read the {@code header} from the given {@link LineIterator} for the GENCODE GTF File.
+     * Will also validate this {@code header} for correctness before returning it.
+     * Throws a {@link UserException.MalformedFile} if the header is malformed.
+     * @param reader The {@link LineIterator} from which to read the header.
+     * @return The header as read from the {@code reader}
+     */
+    private List<String> readActualHeader(LineIterator reader) {
+
+        boolean isFirst = true;
+
+        while ( reader.hasNext() ) {
+            String line = reader.peek();
+
+            // The file will start with commented out lines.
+            // Grab them until there are no more commented out lines.
+            if ( line.startsWith(COMMENT_START) ) {
+                header.add(line);
+                reader.next();
+                isFirst = false;
+            }
+            else if ( isFirst ) {
+                throw new UserException.MalformedFile("GENCODE file does not have a header!");
+            }
+            else {
+                // Validate our header:
+                if ( !validateHeader(header.toArray(new String[0])) ) {
+                    throw new UserException.MalformedFile("Invalid GENCODE GTF File - Header does not conform to GENCODE GTF Specifications!");
+                }
+
+                break;
+            }
+        }
+
+        // Set our line number to be the line of the first actual Feature:
+        currentLineNum = 6;
```

Comment:
Pulling out the number of lines in a valid header and changing assignment to `HEADER_NUM_LINES + 1`